domain/user/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain/user/state"
+)
+
+// InsertAdminUser inserts the admin user into database.
+func InsertAdminUser() func(context.Context, database.TxnRunner) error {
```

Comment:
Assuming this doesn't change based on other PR comments I think we can produce a better name here.

"Insert" is very much a database term. I feel like create or add is a much more reflective term and keeps inline with the services layer.

My proposal would be make this a generic `AddUser` func. Then write a another func called `GenerateAdminUser` that will build up the Admin User with all of the good defaults.

Reply:
I am just thinking about this a little more and I think we are going to need another change here as well. Hopefully I can explain this correctly.

If you look through the DDL for the controller we have a lot of references for `owner` that are not wired up yet. One of the things you are going to need to do soon is start wiring up those foreign keys and this is going to require access to the UUID.

Because of the way our bootstrap process works all of these functions are passed in as ops and so you end up in a chicken and egg situation where you need information that this function doesn't have yet.

My thinking is you could do something like this. I have change the func name based on assumptions but the principal holds true no matter what implementation we end up choosing.

```go
func CreateUser(user User) (user.UUID, func(context.Context, database.TxnRunner) error) {
    // Generate uuid
    return uuid, func
}
```

So now in the agent bootstrap we can call this func above `database.BootstrapControllerConcern` and get the UUID and it can act as input for the rest of the funcs that are going to need it.

Hope my ramblings make sense.

Reply:
I agree with your comment, but what is the problem with generation UUID directly in `user/bootstrap`, without adding so untrivial functions?